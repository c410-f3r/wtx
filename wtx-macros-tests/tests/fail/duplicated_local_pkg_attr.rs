#[wtx_macros::pkg(data_format(json), id(FooId))]
mod pkg {
  #[pkg::aux]
  #[pkg::aux]
  impl Foo {}

  #[derive(Debug)]
  #[pkg::req_data]
  pub struct Req;

  #[pkg::res_data]
  struct Res;
}

#[wtx_macros::pkg(data_format(json), id(FooId))]
mod pkg {
  #[pkg::before_sending]
  #[pkg::before_sending]
  async fn before_sending() -> wtx::Result<()> {
    Ok(())
  }

  #[derive(Debug)]
  #[pkg::req_data]
  pub struct Req;

  #[pkg::res_data]
  struct Res;
}

#[wtx_macros::pkg(data_format(json), id(FooId))]
mod pkg {
  #[derive(Debug)]
  #[pkg::req_data]
  #[pkg::req_data]
  pub struct Req;

  #[pkg::res_data]
  struct Res;
}

fn main() {
}
