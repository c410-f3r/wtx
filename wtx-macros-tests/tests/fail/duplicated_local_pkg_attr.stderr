error: It is not possible to have more than one `pkg` attribute in the same element.
 --> tests/fail/duplicated_local_pkg_attr.rs:3:3
  |
3 |   #[pkg::aux]
  |   ^^^^^^^^^^^

error: It is not possible to have more than one `pkg` attribute in the same element.
  --> tests/fail/duplicated_local_pkg_attr.rs:17:3
   |
17 |   #[pkg::before_sending]
   |   ^^^^^^^^^^^^^^^^^^^^^^

error: It is not possible to have more than one `pkg` attribute in the same element.
  --> tests/fail/duplicated_local_pkg_attr.rs:34:3
   |
34 |   #[pkg::req_data]
   |   ^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

